pkg/metrics/metrics.go

```diff
@@ -0,0 +1,64 @@
+package metrics
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	// MetricsPath is the path that the Metrics are exposed at.
+	MetricsPath = "/metrics"
+
+	// MetricsPort is the port that the Metrics are exposed at.
+	MetricsPort = 8383
+)
+
+var (
+	// Wraps the default RoundTripperFunc with functions that record prometheus metrics.
+	roundTripper promhttp.RoundTripperFunc
+)
+
+// init enables marketplace to serve prometheus metrics.
+func init() {
+	// Register metrics for the operator with the prometheus.
```

Comment:
Question: Is this the pattern recommended or used by prometheus library? As opposed to using a regular initialization function that returns `error`, have it called from `main.go` and handle the error appropriately there?

Reply:
Based on the discussion [here](https://github.com/operator-framework/operator-marketplace/pull/246/files#r321717003), we will no longer exit when failing to register metrics.